### DISCUSSION TITLE
wire:navigate issue

### DISCUSSION THREAD

**Discussion by Minhazul-Islam18** [livewire/livewire#6584] [1. Help] [answered]
in my project, when I changed the page via wire:navigate, some of the functionalities and icons did not work as they should have!

And they throw some error as let and const redeclared! after added **data-navigate-once**(though it only work in first page), errors are disappeared! by functionalities not working!

scripts are in before the end of the body tag like JQuery and others!

After Manual refresh:
![Screenshot 2023-08-30 at 01-19-24 Laravel](https://github.com/livewire/livewire/assets/85400832/1e8a58a5-cfa2-4ff8-b2e7-4bb61b911941)

Via `wire:navigate`:
![Screenshot 2023-08-30 at 01-19-10 Laravel](https://github.com/livewire/livewire/assets/85400832/da297751-3ba4-4eb3-ae73-fc3cee3f5844)

**Comment by Minhazul-Islam18** [comment]
Someone Help me!

**Comment by PhiloNL** [comment]
Hi @Minhazul-Islam18 
It's very difficult to help without any code to replicate the issue and step-by-step instructions on how to trigger the possible bug. If you believe this to be a bug, submitting a failing test would help a lot. Thanks!

**Reply by Minhazul-Islam18** [reply]
Thanks for your reply! I didn't do anything fancy! Just add wire:navigate added to links in my sidebar!
After Navigate to a page using wire:navigate, Then in Console says:
<img width="951" alt="2023-08-30_190457" src="https://github.com/livewire/livewire/assets/85400832/57fb08ee-13b2-4196-9396-cff462c17e7f">


Now, What do you need to see in the code? Please let me know! I'm waiting for your response!

**Reply by Minhazul-Islam18** [reply]
@PhiloNL 
Dear Brother! Can you please help me?

It's come out as an issue when a page is rerendered and scripts are also! then errors are made as let and consts are doubled as variable!

**Reply by PhiloNL** [reply]
Without any code examples it's difficult to help. It could be related to your project for example. If you can create a public repository with a clean installation of Laravel and Livewire and the necessary code that replicates this issue and share this here that would be of great help.

**Comment by N1ebieski** [comment]
wire:navigate is currently bugged a lot. I also reported couple of bugs that still have not been fixed.

Consider disabling this feature in your application at this time.

**Comment by si-aji** [comment, marked as answer]
As my understanding when using wire:navigate, the browser did'nt load new page. But instead it's just changing our current page with new HTML. It's something like

```javascript
document.querySelector('body').innerHTML = newHtml;
```

So if you already declare `let search = null;` on the first 'page' / 'load', the browser already registered search as javascript variable. Then when you're navigating with wire:navigate (or update Livewire model, or anything that trigger Livewire re-render), Livewire will re-render your new HTML, but the browser already know about `search` variable, so they throw an error.

It's related to my discussion  #6515, when the browser already register my event listener. And then when i'm navigating away, the browser still calling my event (generating new string) even though the function is not exists in new HTML.

For myself, I'm doing it this way

```javascript
if(typeof search === 'undefined'){
    // Browser still didn't know about search variable, then register it
    let search = null
}
```

So `search` variable is only registered once in the browser. I still didn't know the overall effect when you're navigating to other 'page'. But it's working for me
